### ISSUE TITLE
[Wishlist] Merge treebrowser plugin and file browser.

### ISSUE THREAD

**Issue by sagarchalise** [geany/geany#1485]
It would make sense to give tree browsing option in current file browser plugin. It seems [treebrowser plugin](https://github.com/geany/geany-plugins/tree/master/treebrowser) currently doesnot have maintainer so it may be possible to merge these two plugins.

**Comment by elextr** [comment]
They are different use-cases, filebrowser is to find files to use in Geany, treebrowser is a second rate file manager that will never be as useful as the system provided one, in other words it doesn't have a use-case, learn to use your system properly `</rant>` :)

In case you didn't catch the drift, I don't favour integrating it.

**Comment by vfaronov** [comment]
@elextr

> treebrowser is a second rate file manager

Or you could ignore its file management features and use it to... you know... find files to use in Geany

**Comment by elextr** [comment]
On 2 May 2017 at 22:07, Vasiliy Faronov <notifications@github.com> wrote:

> @elextr <https://github.com/elextr>
>
> treebrowser is a second rate file manager
>
> Or you could ignore its file management features and use it to... you
> know... find files to use in Geany
>

​Well then it doesn't add anything to the core `filebrowser` plugin, so no
merge is needed.​

**Comment by vfaronov** [comment]
@elextr 

> Well then it doesn't add anything to the core `filebrowser`

Um, the tree thing?

Files in a project are very often spread over multiple directories. It’s very convenient to have them all accessible at the same time. (Especially with #1466.)

**Comment by elextr** [comment]
@vfaronov sounds like you want to take over the maintainership of treebrowser (he says hopefully) ;-D  

Just post a PR on G-P changing the maintainers list.

Then you can apply #1466 to it too.

**Comment by sagarchalise** [comment]
@elextr  I still think a list view option for file browsing would be nice to have in inbuilt plugin. Indeed `treebrowser` and `filebrowser` may have different use case.

**Comment by elextr** [comment]
@sagarchalise yes I believe they have different use-cases, and its my opinion that the file manager part of tree browser is inappropriate for a plugin in an IDE.  If it was part of a refactoring plugin it would make sense, but not simply for file management.

If it was just a browser as its name says, it would still fit the use-case of @vfaronov of having many directories visible at the same time, and in that role could replace file browser so there is not a confusing proliferation of core plugins.  But it should be developed in that role within G-P until shown to be safe and reliable, not moved to core as an unmaintained unknown quantity.

Also if it was moved to core #1414 would need to be updated to include it.

**Comment by codebrainz** [comment]
It would be nice to have something like [SublimeText has](http://shoogledesigns.com/blog/wp-content/uploads/2012/09/switchproject2.jpg) which integrates both the (IMO useless) Documents tab as well as a basic File Browser built-in (to replace Documents tab). It's super handy to have basic file browser/management _integrated_ into the development environment to quickly re-arrange and rename files without doing a full context switch into the desktop's (usually lousy) file manager.

I +1 any effort to do this as I would use it heavily, and if the fact that every other IDE has it builtin is any indication, so would a lot of other people.

**Comment by vfaronov** [comment]
@codebrainz 

> I +1 any effort to do this as I would use it heavily

Just to clarify: why do you not use TreeBrowser right now?

**Comment by codebrainz** [comment]
@vfaronov I usually use ProjectOrganizer which provides some of that functionality (though not nearly enough, IMO). Since the sidebar can only have one thing visible at a time, I usually only keep one plugin loaded which uses the sidebar at a time, and just live with the 2000px of wasted space on the left of Geany.
